crates/nargo_cli/src/cli/prove_cmd.rs

```diff
@@ -106,8 +113,12 @@ pub(crate) fn prove_with_path<B: Backend, P: AsRef<Path>>(
         preprocessed_program;
 
     // Parse the initial witness values from Prover.toml
-    let (inputs_map, _) =
-        read_inputs_from_file(&program_dir, PROVER_INPUT_FILE, Format::Toml, &abi)?;
+    let (inputs_map, _) = read_inputs_from_file(
+        &program_dir,
+        prover_name.as_deref().unwrap_or(PROVER_INPUT_FILE),
```

Comment:
Pretty sure that we can set up clap so we don't need this explicit `unwrap_or` here. Would need to check but we can likely remove the `Option` around the string as we have a default value.